src/Compilers/CSharp/Test/Emit3/Semantics/CollectionExpressionArgumentTests.cs

```diff
@@ -0,0 +1,1553 @@
+                  .maxstack  1
+                  IL_0000:  call       "U System.Activator.CreateInstance<U>()"
+                  IL_0005:  ret
+                }
+                """;
+            verifier.VerifyIL("Program.NoArgs<T, U>", expectedIL);
+            verifier.VerifyIL("Program.EmptyArgs<T, U>", expectedIL);
+
+            string sourceB2 = """
+                class Program
+                {
+                    static void Main()
+                    {
+                        NoArgs<int, MyCollection<int>>().Report();
+                        EmptyArgs<int, MyCollection<int>>().Report();
+                    }
+                    static U NoArgs<T, U>()
+                        where U : struct, IAdd<T>
+                    {
+                        return [];
+                    }
+                    static U EmptyArgs<T, U>()
+                        where U : struct, IAdd<T>
+                    {
+                        return [with()];
+                    }
+                }
+                """;
+            verifier = CompileAndVerify(
+                [sourceA, sourceB2, s_collectionExtensions],
+                verify: Verification.Skipped,
+                expectedOutput: IncludeExpectedOutput("[], [], "));
+            verifier.VerifyDiagnostics();
+            expectedIL = """
+                {
+                  // Code size        6 (0x6)
+                  .maxstack  1
+                  IL_0000:  call       "U System.Activator.CreateInstance<U>()"
+                  IL_0005:  ret
+                }
+                """;
+            verifier.VerifyIL("Program.NoArgs<T, U>", expectedIL);
+            verifier.VerifyIL("Program.EmptyArgs<T, U>", expectedIL);
+        }
+
+        [Fact]
+        public void Arguments_TypeParameter()
+        {
+            string sourceA = """
+                using System.Collections;
+                using System.Collections.Generic;
+                interface IAdd<T> : IEnumerable<T>
+                {
+                    void Add(T t);
+                }
+                """;
+            string sourceB = """
+                class Program
+                {
+                    static void NonEmptyArgsNew<T, U>(T t)
+                        where U : IAdd<T>, new()
+                    {
+                        U x;
+                        x = [with(t), t];
+                        x = [t, with(t)];
+                    }
+                    static void NonEmptyArgsStruct<T, U>(T t)
+                        where U : struct, IAdd<T>
+                    {
+                        U y;
+                        y = [with(t), t];
+                        y = [t, with(t)];
+                    }
+                }
+                """;
+            var comp = CreateCompilation([sourceA, sourceB]);
+            comp.VerifyEmitDiagnostics(
+                // (7,13): error CS0417: 'U': cannot provide arguments when creating an instance of a variable type
```

Comment:
I'll be honest: I've reread this error 5 times and still have no idea what it's tell me. Is the issue that we can't find an appropriate constructor?

Reply:
This is the error reported when constructing an instance of the type parameter with arguments: `new U(t)`.

Reply:
Good lord that error is awful.